Make AEM tests fully work on VP4670 UEFI

SergiiDmytruk opened the pull request:
A couple of updates for VP4670.  openQA didn't match a bootloader needle when AEM was installed and didn't wait long enough for AEM Xen to start.

The tests should pass with binaries from https://github.com/TrenchBoot/grub/pull/30 and https://github.com/TrenchBoot/xen/pull/23.

SergiiDmytruk committed:
needles/aem_hw-bootloader-3-20250323: add

Nothing matches GRUB on vp4670 with AEM installed (looks like font
difference).

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

SergiiDmytruk committed:
tests/aem_hw.pm: wait luks-prompt for 120s

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

SergiiDmytruk committed:
tests/aem_hw.pm: extract lukspass_no_video_workaround()

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

krystian-hebel commented on needles/aem_hw-bootloader-3-20250323.png:
This is only for reinstalling on platform with AEM already installed, right?

SergiiDmytruk replied:
Yes, on first installation `firstboot-bootloader-20230913` needle matches, but I went for matching help text anyway in the new needle to not depend on menu's context.

krystian-hebel commented on tests/aem_hw.pm:
```diff
@@ -489,19 +489,9 @@ sub setup_aem {
-    assert_screen "luks-prompt", timeout => 30;
+    lukspass_no_video_workaround("luks-prompt");
+    # AEM boot on Intel platforms can take significant amount of time.
+    assert_screen "luks-prompt", timeout => 120;
```
Commit message says 90s, please update the commit message.

SergiiDmytruk replied:
It was 90s at first, updated: https://github.com/3mdeb/openqa-tests-qubesos/pull/13/commits/eeac643e710d3392c68d73f9b0689a8e750d2d85

krystian-hebel commented on tests/aem_hw.pm:
```diff
@@ -512,22 +502,30 @@ sub wait_for_startup {
+            last;
+        }
+        send_key 'esc';
+        send_key 'esc';
```
I planned to add a second `check_screen($needle, timeout => 1` between those two keys, it already happened to me twice that the platform ended up in text mode after both key presses. Could you add it, since you're reworking it anyway?

SergiiDmytruk replied:
Added in ec372eb1b0c6052e23b7d3b18254b3e972ff85f6.

krystian-hebel replied:
Not quite what I had in mind, the second `esc` should be pressed conditionally, only if there is no match. I thought that the window is too small to cause any problems, but practice showed otherwise.

SergiiDmytruk replied:
https://github.com/3mdeb/openqa-tests-qubesos/compare/ec372eb1b0c6052e23b7d3b18254b3e972ff85f6..d1a36f62d5f525d064f1b975f9410fc9e1eba6c2

SergiiDmytruk committed:
tests/aem_hw.pm: more reliable lukspass_no_video_workaround

Wait for the target needle after hitting escape the first time,
otherwise one of them can be skipped leaving text mode active.

Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>
